WcaOnRails/app/models/competition.rb

```diff
@@ -1771,7 +1771,8 @@ def update_persons_wcif!(wcif_persons, current_user)
     wcif_persons.each do |wcif_person|
       local_assignments = []
       registration = registrations.find { |reg| reg.user_id == wcif_person["wcaUserId"] }
-      # If no registration is found, assume that this is a non-competing staff member being added.
+      # If no registration is found, and the Registration is marked as non-competing, add this person as a non-competing staff member.
+      next unless registration || !wcif_person["registration"] || wcif_person["registration"]["is_competing"] != false
```

Comment:
Rails has a nifty shorthand to make this more readable: `registration.present?` works even on `nil` values and essentially returns whether a value is nil or not (plus some other very arcane ActiveRecord edge cases which are not relevant here anyways)

Reply:
Done.